src/tasks/WasmAppBuilder/WasmAppBuilder.cs

```diff
@@ -259,28 +171,35 @@ protected override bool ExecuteInternal()
             {
                 if (UseWebcil)
                 {
-                    assemblyPath = Path.Combine(asmRootPath, Path.ChangeExtension(Path.GetFileName(assembly), Utils.WebcilInWasmExtension));
+                    assemblyPath = Path.Combine(frameworkPath, Path.ChangeExtension(Path.GetFileName(assembly), Utils.WebcilInWasmExtension));
                     // For the hash, read the bytes from the webcil file, not the dll file.
                     bytes = File.ReadAllBytes(assemblyPath);
                 }
 
-                config.Assets.Add(new AssemblyEntry(Path.GetFileName(assemblyPath), Utils.ComputeIntegrity(bytes)));
+                config.resources.assembly[Path.GetFileName(assemblyPath)] = Utils.ComputeIntegrity(bytes);
                 if (DebugLevel != 0)
                 {
-                    var pdb = assembly;
-                    pdb = Path.ChangeExtension(pdb, ".pdb");
+                    var pdb = Path.ChangeExtension(assembly, ".pdb");
                     if (File.Exists(pdb))
-                        config.Assets.Add(new PdbEntry(Path.GetFileName(pdb), Utils.ComputeIntegrity(pdb)));
+                    {
+                        if (config.resources.pdb == null)
+                            config.resources.pdb = new();
+
+                        config.resources.pdb[Path.GetFileName(pdb)] = Utils.ComputeIntegrity(pdb);
+                    }
                 }
             }
         }
 
-        config.DebugLevel = DebugLevel;
+        config.debugBuild = DebugLevel > 0;
```

Comment:
Do we pass in the debug level anywhere now? it will control the log level from the runtime, for example, when `<0`.

Reply:
The value comes from `WasmDebugLevel` property https://github.com/dotnet/runtime/blob/7d2a4d298a3d9bed50089eb79f67e5e8f3b0b190/src/mono/wasm/build/WasmApp.targets#L401

Reply:
But it is getting reduced to a bool here, thus losing the *level* which we would have otherwise passed to the runtime.

Reply:
I see, I'll check that, thanks!

Reply:
I have added a new property in boot config